trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/SigningHeaders.java

```diff
@@ -76,25 +69,30 @@ boolean hasHeaderToSign(String name)
 
     Stream<Map.Entry<String, List<String>>> lowercaseHeadersToSign()
     {
-        return lowercaseHeaders.entrySet().stream()
+        return headers.entrySet().stream()
                 .filter(entry -> lowercaseHeadersToSign.contains(entry.getKey()));
     }
 
     Optional<String> getFirst(String lowercaseHeader)
     {
-        return Optional.ofNullable(lowercaseHeaders.getFirst(lowercaseHeader));
+        return Optional.ofNullable(headers.getFirst(lowercaseHeader));
     }
 
-    private static MultivaluedMap<String, String> buildLowercaseHeaders(SigningController.Mode mode, MultivaluedMap<String, String> headers)
+    private static String lowercaseHeader(String headerName, String headerValue)
     {
-        BiFunction<String, List<String>, List<String>> lowercaseHeaderValues = (key, values) -> {
-            // mode == UNADJUSTED_HEADERS is temp until airlift is fixed
-            if ((mode == UNADJUSTED_HEADERS) || !LOWERCASE_HEADERS.contains(key)) {
-                return values;
-            }
-            return values.stream().map(value -> value.toLowerCase(Locale.ROOT)).collect(toImmutableList());
-        };
+        if (!LOWERCASE_HEADERS.contains(headerName)) {
+            return headerValue;
+        }
+        return headerValue.toLowerCase(Locale.ROOT);
+    }
 
-        return lowercase(headers, lowercaseHeaderValues);
+    private static MultiMap adjustHeaders(SigningController.Mode mode, MultiMap headers)
+    {
+        if (mode == UNADJUSTED_HEADERS) {
```

Comment:
This mode will go away very soon. It might be best to leave this hidden in `Signer`.